openghg/util/_domain.py

```diff
@@ -3,21 +3,49 @@
 import numpy as np
 from numpy import ndarray
 
+from openghg.types import optionalPathType
 
-def find_domain(domain: str) -> Tuple[ndarray, ndarray]:
+__all__ = ["get_domain_info", "find_domain", "convert_longitude"]
+
+
+def get_domain_info(domain_filename: optionalPathType = None) -> Dict[str, Any]:
+    """
+    Extract data from domain info JSON file as a dictionary.
+
+    This uses the data stored within openghg_defs/domain_info JSON file by default.
+
+    Args:
+        domain_filename: Alternative domain info file.
+
+    Returns:
+        dict: Data from domain JSON file
+    """
+    from openghg_defs import domain_info_file
+    from openghg.util import load_json
+
+    if domain_filename is None:
+        domain_info_json = load_json(domain_info_file)
+    else:
+        domain_info_json = load_json(domain_info_file)
+
+    return domain_info_json
+
+
+def find_domain(domain: str,
+                domain_filename: optionalPathType = None) -> Tuple[ndarray, ndarray]:
     """
     Finds the latitude and longitude values in degrees associated
     with a given domain name.
 
     Args:
-        domain: Pre-defined domain name (see 'domain_info.json')
+        domain: Pre-defined domain name
+        domain_filename: Alternative domain info file. Defaults to openghg_defs input.
 
     Returns:
         array, array : Latitude and longitude values for the domain in degrees.
     """
-    from openghg.util import load_json
 
-    domain_info = load_json(filename="domain_info.json")
+    domain_info = get_domain_info(domain_filename)
```

Comment:
Add in keyword before argument

Reply:
As this only has one input (`domain_filename`) is this necessary do you think? And also wouldn't that lock us into that variable name when we don't necessarily need to?